packages/shared/src/components/modals/SquadMemberModal.tsx

```diff
@@ -49,21 +49,23 @@ const InitialItem = ({ squad }: { squad: Squad }) => {
 export function SquadMemberModal({
   squad,
   ...props
-}: UpvotedPopupModalProps): ReactElement {
-  const { user: loggedUser } = useAuthContext();
+}: SquadMemberModalProps): ReactElement {
+  const [roleFilter, setRoleFilter] = useState<SourceMemberRole>(null);
   const [member, setMember] = useState<SourceMember>(null);
   const { onMenuClick } = useContextMenu({ id: 'squad-member-menu-context' });
   const {
     members,
     membersQueryResult: queryResult,
+    onUnblock,
     onUpdateRole,
     verifyPermission,
   } = useSquadActions({
     squad,
+    membersQueryParams: { role: roleFilter },
```

Comment:
One side of me likes this, other side feels like it's a bit hacky around the modal guideline.
Not sure if we should change or not, maybe @capJavert has an opinion here?

Reply:
I agree with your sentiment. One part makes it quick and working, on the other side, we are not really "switching" the UI per tab as the guideline would suggest. I am willing to refactor if we decide to do two separate components for the list.

Reply:
Sorry, missed this, what guideline are we talking about specifically?

Reply:
Mainly modal guideline, normally the stepped modal expect 2 seperate views.
Here we technically "hacked" a workaround for updates outside the modal.

Reply:
If in all other places we do it with two views then this one should be no exception. But don't have the hard preference because above is simple logic wise.